File: coolrmi/hu.qgears.coolrmi/src/hu/qgears/coolrmi/messages/CoolRMIRequestServiceQuery.java
package hu.qgears.coolrmi.messages;

import java.io.Serializable;

import hu.qgears.coolrmi.serializer.PortableSerializable;


public class CoolRMIRequestServiceQuery
	extends AbstractCoolRMIMessage
	implements Serializable, PortableSerializable {
	private static final long serialVersionUID = 1L;
	private String serviceName;
	public CoolRMIRequestServiceQuery() {
		
	}

	public CoolRMIRequestServiceQuery(long queryId, String serviceName) {
		super(queryId);
		this.serviceName = serviceName;
	}

	public String getServiceName() {
		return serviceName;
	}
	@Override
	public String toString() {
		return "Request service "+serviceName+" "+getQueryId();
	}

	@Override
	public String getName() {
		return toString();
	}
}

File: coolrmi/CoolRMI.Net/Messages/CoolRMIRequestServiceQuery.cs
using System;
using CoolRMI.Net.Serializer;

namespace CoolRMI.Net.Messages
{
    [Serializable]
    public class CoolRMIRequestServiceQuery : AbstractCoolRMIMessage
    {
        [PortableFieldName("serviceName")]
        public string ServiceName { get; }
        public override string Name => ToString();

        public CoolRMIRequestServiceQuery(long queryId, string serviceName) :
            base(queryId)
        {
            ServiceName = serviceName;
        }

        public override string ToString()
        {
            return "Request service " + ServiceName + " " + QueryId;
        }
    }

    [Serializable]
    public class CoolRMIRequestServiceReply : AbstractCoolRMIReply
    {
        [PortableFieldName("proxyId")]
        public long ProxyId { get; }
        [PortableFieldName("interfaceName")]
        public string InterfaceName { get; }
        public override string Name => ToString();

        public CoolRMIRequestServiceReply(long queryId, long proxyId,
            string interfaceName) : base(queryId)
        {
            ProxyId = proxyId;
            InterfaceName = interfaceName;
        }

        public override string ToString()
        {
            return "Request service reply";
        }
    }
}
